keep/api/api.py

```diff
@@ -159,84 +161,100 @@ def get_app(
+        app.include_router(users.router, prefix="/users", tags=["users"])
+        app.include_router(topology.router, prefix="/topology", tags=["topology"])
+        app.include_router(
+            mapping.router, prefix="/mapping", tags=["enrichment", "mapping"]
+        )
+        app.include_router(
+            extraction.router, prefix="/extraction", tags=["enrichment", "extraction"]
+        )
+        app.include_router(dashboard.router, prefix="/dashboard", tags=["dashboard"])
+        app.include_router(tags.router, prefix="/tags", tags=["tags"])
+
+        # If we run Keep with SINGLE_TENANT auth type, we want to add the signin endpoint
+        if AUTH_TYPE == AuthenticationType.SINGLE_TENANT.value:
+
+            @app.post("/signin")
+            def signin(body: dict):
+                # validate the user/password
+                user = get_user(body.get("username"), body.get("password"))
+
+                if not user:
+                    return JSONResponse(
+                        status_code=401,
+                        content={"message": "Invalid username or password"},
+                    )
+                # generate a JWT secret
+                jwt_secret = os.environ.get("KEEP_JWT_SECRET")
+                if not jwt_secret:
+                    logger.info("missing KEEP_JWT_SECRET environment variable")
+                    raise HTTPException(status_code=401, detail="Missing JWT secret")
+                token = jwt.encode(
+                    {
+                        "email": user.username,
+                        "keep_tenant_id": SINGLE_TENANT_UUID,
+                        "role": user.role,
+                    },
+                    jwt_secret,
+                    algorithm="HS256",
                 )
-            # generate a JWT secret
-            jwt_secret = os.environ.get("KEEP_JWT_SECRET")
-            if not jwt_secret:
-                logger.info("missing KEEP_JWT_SECRET environment variable")
-                raise HTTPException(status_code=401, detail="Missing JWT secret")
-            token = jwt.encode(
-                {
+                # return the token
+                return {
+                    "accessToken": token,
+                    "tenantId": SINGLE_TENANT_UUID,
                     "email": user.username,
-                    "keep_tenant_id": SINGLE_TENANT_UUID,
                     "role": user.role,
-                },
-                jwt_secret,
-                algorithm="HS256",
-            )
-            # return the token
-            return {
-                "accessToken": token,
-                "tenantId": SINGLE_TENANT_UUID,
-                "email": user.username,
-                "role": user.role,
-            }
+                }
+
+    # WORKER MODE
+    elif mode == "worker":
+        # Initialize only the worker-related components
+        @app.on_event("startup")
+        async def start_worker():
+            if REDIS and WORKER_ENABLED:
```

Comment:
Looks like this PR is making WORKER_ENABLED env var obsolete, it used to serve the same purpose:
- REDIS & WORKER_ENABLED for worker
- REDIS & !WORKER_ENABLED for api only